ocsp.go

```diff
@@ -720,7 +720,7 @@ func initOCSPCache() {
 	ocspResponseCacheLock = &sync.RWMutex{}
 
 	logger.Infof("reading OCSP Response cache file. %v\n", cacheFileName)
-	f, err := os.Open(cacheFileName)
+	f, err := os.OpenFile(cacheFileName, os.O_CREATE|os.O_RDONLY, os.ModePerm)
```

Comment:
Should this be created in write mode?

Reply:
Also please change this to a warning. We shouldn't be throwing errors if ocsp cache dosn't exist locally..

Reply:
I believe we are trying to read from the file.

Reply:
Sounds good. I'd still suggest update the loggere.Error to something similar to what Python has - https://github.com/snowflakedb/snowflake-connector-python/blob/f168a26a251361bac2fd8822636085e7e20132a9/src/snowflake/connector/ocsp_snowflake.py#L501

Reply:
Here was my concern when I commented on this - https://golang.org/pkg/os/#example_OpenFile .

> If the file does not exist, and the O_CREATE flag is passed, it is created with mode perm (before umask)

Is it possible that a file is created with READONLY permissions which then the code later cannot write to?

Reply:
Nope, it just means that it returns a reader such that we read contents from it, and cannot write to it using the same file pointer. If we want to write to it, we can open the file again with a WRITEONLY pointer. In fact, we can pass in a READ/WRITE flag to do both, but I didn't see the point of having a write flag we wouldn't use.

Reply:
Makes sense.